mux/src/tmux_commands.rs

```diff
@@ -255,14 +674,267 @@ impl TmuxCommand for ListAllPanes {
                 pane_height,
                 pane_left,
                 pane_top,
+                pane_active,
+            });
+        }
+
+        log::debug!("panes in domain_id {}: {:?}", domain_id, items);
+        let mux = Mux::get();
+        if let Some(domain) = mux.get_domain(domain_id) {
+            if let Some(tmux_domain) = domain.downcast_ref::<TmuxDomain>() {
+                if !self.prune {
+                    return tmux_domain.inner.sync_pane_state(&items);
+                } else {
+                    return tmux_domain
+                        .inner
+                        .remove_detached_pane(self.window_id, &pane_set);
+                }
+            }
+        }
+        anyhow::bail!("Tmux domain lost");
+    }
+}
+
+fn parse_pane(layout: &str) -> anyhow::Result<PaneItem> {
```

Comment:
all the unwraps and expects in here need to be converted to error propagation so that wezterm cannot die just because it got something it didn't understand from tmux!